.github/workflows/rubyonrails.yml

```diff
@@ -55,6 +55,61 @@ jobs:
       - name: Run tests
         run: bin/rails test
 
+  system_tests:
+    runs-on: ubuntu-latest
+    container:
+      image: ruby:3.2.2
+    services:
+      postgres:
+        image: postgres:16.1
+        ports:
+          - "5432:5432"
+        env:
+          POSTGRES_DB: hostedgpt_test
+          POSTGRES_USER: rails
+          POSTGRES_PASSWORD: password
+    env:
+      RAILS_ENV: test
+      DATABASE_URL: "postgres://rails:password@postgres:5432/hostedgpt_test"
+    steps:
+      - run: |
+          set -x
+          apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 4EB27DB2A3B88B8B
+          echo "deb [arch=amd64] https://dl.google.com/linux/chrome/deb/ stable main" | tee /etc/apt/sources.list.d/google.list >/dev/null
+          apt-get update
+          apt-get install -y google-chrome-stable jq curl unzip
+
+      - run: |
+          export DISPLAY=:99
+          chromedriver --url-base=/wd/hub &
+
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Cache gems
+        uses: actions/cache@v3
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-gems-
+
+      - name: Install Bundler
+        run: |
+          gem update --system
+          gem install bundler
+
+      - name: Install gems
+        run: |
+          bundle config set path vendor/bundle
+          bundle install
+
+      - name: Set up database schema
+        run: bin/rails db:schema:load
+
```

Comment:
I don't think you need this; I was pretty sure tests did this automatically at the start of each test run, but I could be wrong. There is little harm in having it.

Reply:
I was surprised to see this in here too -- I just copied most of this action from the workflow that was already present.